WordPress/src/main/java/org/wordpress/android/ui/ShareIntentReceiverActivity.java

```diff
@@ -4,233 +4,158 @@
+ * along with the content passed in the intent.
  */
-public class ShareIntentReceiverActivity extends AppCompatActivity implements OnItemSelectedListener {
-    public static final String SHARE_LAST_USED_BLOG_ID_KEY = "wp-settings-share-last-used-text-blogid";
-    public static final String SHARE_LAST_USED_ADDTO_KEY = "wp-settings-share-last-used-image-addto";
+public class ShareIntentReceiverActivity extends AppCompatActivity implements ShareIntentFragmentListener {
 
-    private static final int ADD_TO_NEW_POST = 0;
-    private static final int ADD_TO_MEDIA_LIBRARY = 1;
+    public static final String SHARE_LAST_USED_BLOG_ID_KEY = "wp-settings-share-last-used-text-blogid";
+    private static final String KEY_SELECTED_SITE_LOCAL_ID = "KEY_SELECTED_SITE_LOCAL_ID";
+    private static final String KEY_SHARE_ACTION_ID = "KEY_SHARE_ACTION_ID";
 
     @Inject AccountStore mAccountStore;
     @Inject SiteStore mSiteStore;
 
-    private Spinner mBlogSpinner;
-    private RadioGroup mActionGroup;
-    private String mSiteNames[];
-    private int mSiteIds[];
-    private int mSelectedSiteLocalId;
-    private int mActionIndex;
+    private int mClickedSiteLocalId;
+    private String mShareActionName;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         ((WordPress) getApplication()).component().inject(this);
+        setContentView(R.layout.share_intent_receiver_activity);
 
-        setContentView(R.layout.share_intent_receiver_dialog);
-
-        TextView blogSpinnerTitle = (TextView) findViewById(R.id.blog_spinner_title);
-        mBlogSpinner = (Spinner) findViewById(R.id.blog_spinner);
-        mActionGroup = (RadioGroup) findViewById(R.id.share_actions);
-
-        initSiteLists();
-
-        if (mSiteNames == null) {
-            finishIfNoVisibleBlogs();
-            return;
-        }
-
-        if (mSiteNames.length == 1) {
-            mBlogSpinner.setVisibility(View.GONE);
-            blogSpinnerTitle.setVisibility(View.GONE);
-        } else {
-            ArrayAdapter<String> adapter = new ArrayAdapter<>(this, R.layout.spinner_menu_dropdown_item, mSiteNames);
-            mBlogSpinner.setAdapter(adapter);
-            mBlogSpinner.setOnItemSelectedListener(this);
-        }
-
-        loadLastUsed();
-
-        // If type is text/plain hide Media Gallery option
-        if (isSharingText()) {
-            mActionIndex = ADD_TO_NEW_POST;
-            mActionGroup.setVisibility(View.GONE);
-            findViewById(R.id.action_spinner_title).setVisibility(View.GONE);
-            // if text/plain and only one blog, then don't show this fragment, share it directly to a new post
-            if (mSiteNames.length == 1) {
-                // Single site, startActivityAndFinish will pick the first one by default
-                startActivityAndFinish(new Intent(this, EditPostActivity.class));
```

Comment:
Looks like this changeset killed a small shortcut here: if the user shares text and have one only site, we didn't show the picker at all but we opened the editor directly.

Reply:
Good point! I must have missed that. Thanks

Reply:
I'm having some troubles with activity transitions when skipping the 'pick site' fragment. The ShareIntentReceiverActivity UI is shown for a short time which looks pretty bad. When the UI was just a dialog, the activity had "NoDisplay" theme so the transition was invisible. 
The only solution I can think of is to create a new activity with NoDisplay theme, which would decide whether to go directly to EditPostActivity or shown the 'pick site' activity.
Is there any better solution? Thanks @maxme

Reply:
N/C :) ... I'm using the TaskStackBuilder and I was setting the FLAG_ACTIVITY_NO_ANIMATION on a wrong intent. It works as expected.

Reply:
It didn't work when the app was dead and a user wanted to share a text. The activity transition was still visible to the user. So I had to use the NoDisplay layout after all and set the 'LoginTheme' directly on the views.